### PR TITLE
do not overwrite unchanged cached Credentials (#1168)

### LINKED ISSUE #1168: Write credentials to cache only if they are changed
**Is your feature request related to a problem? Please describe.**
A minor problem. When login with cached credentials and `store_credentials` is `true` in call to `Session::connect()` the file with cached credentials is written every time `Session::connect()` is called. The contents of the file usually do not change with these writes.

**Describe the solution you'd like**
The file with cached credentials should be written only if cached credentials have changed (have been updated by Spotify).

**Describe alternatives you've considered**
Set `store_credentials` to `false` and use code that writes credentials to cache if they are changed.
https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/blob/ea26d0d6d50bc0fd394c8934b14d18266671264b/audio/spotify/src/common.rs#L162

**Additional context**
Problem was noticed in: https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/merge_requests/1195

### PR DESCRIPTION
**Pull request by jokester**
fixes #1168

### CODE REVIEW

**Comment by jokester** [comment]
This is my first time coding in a real rust project. I didn't even know `Arc<T>` before this day.

If something goes wrong I'm sorry :rofl:

File: playback/src/audio_backend/jackaudio.rs [hunk 1]
```diff
@@ -47,8 +47,8 @@ impl Open for JackSink {
         let client_name = client_name.unwrap_or_else(|| "librespot".to_string());
         let (client, _status) =
             Client::new(&client_name[..], ClientOptions::NO_START_SERVER).unwrap();
-        let ch_r = client.register_port("out_0", AudioOut::default()).unwrap();
-        let ch_l = client.register_port("out_1", AudioOut::default()).unwrap();
+        let ch_r = client.register_port("out_0", AudioOut).unwrap();
```
Comment (jokester) [inline playback/src/audio_backend/jackaudio.rs:50]: This is to fix clippy error in https://github.com/librespot-org/librespot/actions/runs/5544745826

**Comment by roderickvd** [comment]
Looks good to me, thanks!

**Comment by Laiteux** [comment]
@jokester @roderickvd this PR does not work. The logic is not correct and can't be correct anyway as the problem lies elsewhere.

The issue is that right now, no matter what, librespot will reconnect to your account every time you use `Session::connect`.

This is problematic because
1. This might trigger stuff on Spotify's side
2. Indeed, it does, as if done too many times, Spotify will force you a reset password

A token refresh if the current session dies would be the way. Still, the issue remains that the stored `auth_data` isn't being reused at all. Cached but never used.

**Comment by Laiteux** [comment]
Could you use `Session.is_invalid()` to check if the session is dead? or what is that for? Not fluent enough in Rust.

**Comment by oSumAtrIX** [comment]
@Laiteux You may want to open a new issue, as this PR is closed and will unlikely get any attention from devs.

**Pull request merged**